### ISSUE TITLE
Validating datetime strings with timezone produces an error

### ISSUE THREAD

**Issue by ToivoMattilaVirnex** [great-expectations/great_expectations#9203]
**Describe the bug**
Validating datetime strings with timezone produces an error

Using **expect_column_values_to_match_strftime_format** to validate datetime strings (i.e. "2024-01-01") that contain timezone information produces an error.

**To Reproduce**

Below is code for reproducing the error.
The code uses in-memory context and doesn't have a filesystem project or great_expectations.yml.

Code:
```
# Imports
import pandas as pd
import great_expectations as gx
# Set up data
dataframe = pd.DataFrame({"datetime": ["2023-01-01T00:00:00+0100"]})
# Set up GX
context = gx.get_context()
datasource = context.sources.add_pandas(name="my_pandas_datasource")
data_asset = datasource.add_dataframe_asset(name="my_data_asset")
batch_request = data_asset.build_batch_request(dataframe=dataframe)
context.add_or_update_expectation_suite("my_expectation_suite")
# Set up validation
validator = context.get_validator(
    batch_request=batch_request,
    expectation_suite_name="my_expectation_suite",
)
# This produces the error
validator.expect_column_values_to_match_strftime_format(
    column="datetime",
    strftime_format="%Y-%m-%dT%H:%M:%S%z"
)
```

Running this code produces an error
The traceback from the error is below:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File /mnt/d/code/gx_timezone_bug/.venv/lib/python3.10/site-packages/great_expectations/expectations/core/expect_column_values_to_match_strftime_format.py:184, in ExpectColumnValuesToMatchStrftimeFormat.validate_configuration(self, configuration)
    183     else:
--> 184         datetime.strptime(  # noqa: DTZ007
    185             datetime.strftime(datetime.now(), strftime_format),  # noqa: DTZ005
    186             strftime_format,
    187         )
    188 except ValueError as e:

File /usr/lib/python3.10/_strptime.py:568, in _strptime_datetime(cls, data_string, format)
    566 """Return a class cls instance based on the input string and the
    567 format string."""
--> 568 tt, fraction, gmtoff_fraction = _strptime(data_string, format)
    569 tzname, gmtoff = tt[-2:]

File /usr/lib/python3.10/_strptime.py:349, in _strptime(data_string, format)
    348 if not found:
--> 349     raise ValueError("time data %r does not match format %r" %
    350                      (data_string, format))
    351 if len(data_string) != found.end():

ValueError: time data '2024-01-09T14:52:59' does not match format '%Y-%m-%dT%H:%M:%S%z'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
Cell In[9], line 1
----> 1 validator.expect_column_values_to_match_strftime_format(
      2     column="datetime",
      3     strftime_format="%Y-%m-%dT%H:%M:%S%z"
      4 )

File /mnt/d/code/gx_timezone_bug/.venv/lib/python3.10/site-packages/great_expectations/validator/validator.py:591, in Validator.validate_expectation.<locals>.inst_expectation(*args, **kwargs)
    585         validation_result = ExpectationValidationResult(
    586             success=False,
    587             exception_info=exception_info,
    588             expectation_config=configuration,
    589         )
    590     else:
--> 591         raise err
    593 if self._include_rendered_content:
    594     validation_result.render()

File /mnt/d/code/gx_timezone_bug/.venv/lib/python3.10/site-packages/great_expectations/validator/validator.py:546, in Validator.validate_expectation.<locals>.inst_expectation(*args, **kwargs)
    539     configuration.process_evaluation_parameters(
    540         self._expectation_suite.evaluation_parameters,
    541         True,
    542         self._data_context,
    543     )
    545 try:
--> 546     expectation = expectation_impl(configuration)
    547     """Given an implementation and a configuration for any Expectation, returns its validation result"""
    549     if not self.interactive_evaluation and not self._active_validation:

File /mnt/d/code/gx_timezone_bug/.venv/lib/python3.10/site-packages/great_expectations/expectations/expectation.py:356, in Expectation.__init__(self, configuration)
    352 def __init__(
    353     self, configuration: Optional[ExpectationConfiguration] = None
    354 ) -> None:
    355     if configuration:
--> 356         self.validate_configuration(configuration=configuration)
    358     self._configuration = configuration

File /mnt/d/code/gx_timezone_bug/.venv/lib/python3.10/site-packages/great_expectations/expectations/core/expect_column_values_to_match_strftime_format.py:189, in ExpectColumnValuesToMatchStrftimeFormat.validate_configuration(self, configuration)
    184         datetime.strptime(  # noqa: DTZ007
    185             datetime.strftime(datetime.now(), strftime_format),  # noqa: DTZ005
    186             strftime_format,
    187         )
    188 except ValueError as e:
--> 189     raise ValueError(f"Unable to use provided strftime_format. {e!s}")
    190 except AssertionError as e:
    191     raise InvalidExpectationConfigurationError(str(e))

ValueError: Unable to use provided strftime_format. time data '2024-01-09T14:52:59' does not match format '%Y-%m-%dT%H:%M:%S%z'
```

**Expected behavior**
The code doesn't produce an error
Instead, it validates that the strings in the column conform to the provided strftime format, including the timezone.

**Environment:**
 - Operating System: Linux (WSL)
 - Great Expectations Version: 0.18.7
 - Python Version: 3.10.12
 - Data Source: Pandas
 - Cloud environment: None

**Comment by molliemarie** [comment]
Hello @ToivoMattilaVirnex. With the launch of Great Expectations Core (GX 1.0), we are closing old issues posted regarding previous versions. Moving forward, we will focus our resources on supporting and improving GX Core (version 1.0 and beyond). If you find that an issue you previously reported still exists in GX Core, we encourage you to resubmit it against the new version. With more resources dedicated to community support, we aim to tackle new issues swiftly. For specific details on what is GX-supported vs community-supported, you can reference our [integration and support policy](https://docs.greatexpectations.io/docs/application_integration_support/).

To get started on your transition to GX Core, check out the [GX Core quickstart](https://docs.greatexpectations.io/docs/core/introduction/try_gx/#test-features-and-functionality) (click “Full example code” tab to see a code example).

You can also join our upcoming community meeting on August 28th at 9am PT (noon ET / 4pm UTC) for a comprehensive rundown of everything GX Core, plus Q&A as time permits. Go to https://greatexpectations.io/meetup and click “follow calendar” to follow the GX community calendar.

Thank you for being part of the GX community and thank you for submitting this issue. We're excited about this new chapter and look forward to your feedback on GX Core. 🤗

**Comment by choinhet** [comment]
I still face the same issue in GX version 1.16

**Comment by joshua-stauffer** [comment]
Thanks for the fix @choinhet - this bug is addressed in [1.16.1](https://github.com/great-expectations/great_expectations/releases/tag/1.16.1)
